iphone/cli/hooks/storyboard.js

```diff
@@ -0,0 +1,591 @@
+/**
+ * iOS build hook that scans for available storyboards from modules
+ * and then configures them in the Xcode project
+ */
+
+// TO DO: Can we merge storyboard.js and farmework.js
+
+'use strict';
+
+const fs = require('fs-extra');
+const IncrementalFileTask = require('appc-tasks').IncrementalFileTask;
+const path = require('path');
+
+exports.cliVersion = '>=3.2.1';
+
+exports.init = function (logger, config, cli) {
+	let storyboardManager = new StoryboardManager(cli, config, logger);
+	storyboardManager.initialize();
+};
+
+/**
+ * Manages all available storyboards from modules.
+ *
+ * Scans for available storyboards and inspects them to get the required info to
+ * properly integrate them into the Xcode project.
+ */
+class StoryboardManager {
+
+	/**
+	 * Constructs a new storyboard manager.
+	 *
+	 * @param {Object} cli - CLI instance
+	 * @param {Object} config - Project configuration
+	 * @param {Object} logger - Logger instance
+	 * @access public
+	 */
+	constructor(cli, config, logger) {
+		this._cli = cli;
+		this._config = config;
+		this._logger = logger;
+		this._builder = null;
+		this._storyboards = new Map();
+	}
+
+	/**
+	 * Initializes the storyboard manager by hooking into the required build steps.
+	 *
+	 * @access public
+	 */
+	initialize() {
+		this._cli.on('build.pre.compile', {
+			priority: 1200,
+			post: (builder, callback) => {
+				this._logger.trace('Starting storyboard detection');
```

Comment:
Please guard all of this to only happen on iOS

Reply:
Is it really required to guard for iOS? This hook is already inside iphone/cli/hooks. I believe it will there only when iOS is building.